src/Symfony/Bundle/SecurityBundle/Security/FirewallAwareTrait.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Security;
+
+/**
+ * Provides basic functionality for services mapped by the firewall name
+ * in a container locator.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @internal
+ */
+trait FirewallAwareTrait
```

Comment:
I'm not sure this is worth the added complexity/duck typing/readability loss

Reply:
I'm happy to revert this change. As there are quite some checks (3 if statements + error messages) and it's used by 3 classes if this PR is merged, I thought it was nice to make a trait for it (to not forget a check).